src/main/java/com/corundumstudio/socketio/handler/ClientsBox.java

```diff
@@ -15,19 +15,51 @@
  */
 package com.corundumstudio.socketio.handler;
 
+import com.corundumstudio.socketio.SocketIOServer;
 import io.netty.channel.Channel;
 import io.netty.util.internal.PlatformDependent;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
 
 import com.corundumstudio.socketio.HandshakeData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ClientsBox {
+    private static final Logger log = LoggerFactory.getLogger(ClientsBox.class);
 
     private final Map<UUID, ClientHead> uuid2clients = PlatformDependent.newConcurrentHashMap();
     private final Map<Channel, ClientHead> channel2clients = PlatformDependent.newConcurrentHashMap();
 
+    private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+
+    public ClientsBox() {
+        scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            List<UUID> disconnected = new ArrayList<>();
+            for (Map.Entry<UUID, ClientHead> entry : uuid2clients.entrySet()) {
+                if (!entry.getValue().isConnected()) {
+                    disconnected.add(entry.getKey());
+                }
+            }
+            try {
+                TimeUnit.SECONDS.sleep(3);
```

Comment:
Can we avoid this sleep call? Don't want to have sleeping threads

Reply:
It's ok